### PR TITLE
Implement Task Router API

### PR DESCRIPTION
**Pull request by enilsen16**
This is most likely going to take me a few days to implement.
- Resources
  - [x] Workspaces
  - [x] Workers
  - [x] Activities
  - [x] TaskQueues
  - [x] Workflows
  - [x] Tasks
  - [x] Statistics
  - [x] Events
- [x] Add documentation
  - [x] Squash Commits
- [x] Update Readme

Let me know if there's anything you think I am missing @danielberkompas.

### CODE REVIEW

**Comment by enilsen16** [comment]
@danielberkompas If I understand correctly `parent/1` is for adding into the path before the resource and `children/1` is to add after the resource. Is that correct?

**Comment by danielberkompas** [comment]
@enilsen16 yes, that's correct.

**Comment by enilsen16** [comment]
Alright, I need to go through and change some things before you merge this 😃

File: lib/ex_twilio/url_generator.ex [hunk 1]
```diff
@@ -64,6 +63,17 @@ defmodule ExTwilio.UrlGenerator do
     end
   end
 
+  defp add_segements(url, module, id, options) do
```
Comment (danielberkompas) [inline lib/ex_twilio/url_generator.ex]: Should be `add_segments`

File: lib/ex_twilio/url_generator.ex [hunk 2]
```diff
@@ -158,7 +173,11 @@ defmodule ExTwilio.UrlGenerator do
 
   @spec infer_module(atom) :: atom
   defp infer_module(atom) do
```
Comment (danielberkompas) [inline lib/ex_twilio/url_generator.ex]: Could do this with pattern matching.

```elixir
defp infer_module(:workspace), do: ExTwilio.TaskRouter.Workspace
defp infer_module(atom) do
  Module.concat(ExTwilio, camelize(atom))
end
```

File: lib/ex_twilio/resources/task_router/activity.ex [hunk 3]
```diff
@@ -0,0 +1,21 @@
+defmodule ExTwilio.TaskRouter.Activity do
+  @moduledoc """
+  Repersents the current status of your workers. Workers can only have a single activity at a time.
```
Comment (joshuafleck) [inline lib/ex_twilio/resources/task_router/activity.ex]: `Repersents` -> `Represents`

File: lib/ex_twilio/resources/task_router/workspace.ex [hunk 4]
```diff
@@ -0,0 +1,27 @@
+defmodule ExTwilio.TaskRouter.Workspace do
+  @moduledoc """
+  A Workspace is a container for your Tasks, Workers, TaskQueues, Workflows and Activities.
+  Each of these items exists within a single Workspace and will not be shared across Workspaces.
+
+  - [Twilio docs](https://www.twilio.com/docs/api/taskrouter/workflows)
```
Comment (joshuafleck) [inline lib/ex_twilio/resources/task_router/workspace.ex]: I believe the link should be: `https://www.twilio.com/docs/api/taskrouter/workspaces`

File: lib/ex_twilio/worker_capability.ex [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+  def with_auth_token(capability_struct = %__MODULE__{}, auth_token) do
+    %{capability_struct | auth_token: auth_token}
+  end
+
+  @doc """
+  Sets the Twilio worker sid used to sign the capability token.
+
+  ## Example
+
+  Sets the auth token to be XXX
```
Comment (joshuafleck) [inline lib/ex_twilio/worker_capability.ex]: This should refer to worker sid

File: lib/ex_twilio/worker_capability.ex [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+  def with_worker_sid(capability_struct = %__MODULE__{}, worker_sid) do
+    %{capability_struct | worker_sid: worker_sid}
+  end
+
+  @doc """
+  Sets the Twilio workspace sid used to sign the capability token.
+
+  ## Example
+
+  Sets the auth token to be XXX
```
Comment (joshuafleck) [inline lib/ex_twilio/worker_capability.ex]: This should refer to workspace sid

**Comment by danielberkompas** [comment]
@enilsen16 is this still in progress? Need help with anything?

**Comment by enilsen16** [comment]
I just haven't gotten around to finishing this (working on something else at work). I'll finish it this week though. Thanks @danielberkompas 😄

**Comment by enilsen16** [comment]
@danielberkompas Let me know if there's anything else you like me to add or change. 😄

**Comment by danielberkompas** [comment]
@enilsen16 I'd like to see @joshuafleck's comments addressed and the build passing.

**Comment by enilsen16** [comment]
Updated 😐

Dialyzer seems to be complaining. Any idea how to fix that error @danielberkompas?

Thanks!

**Comment by danielberkompas** [comment]
I'll take care of the dialyzer warnings myself. Thanks! :star:

**Pull request merged**